modules/javafx.graphics/src/main/java/com/sun/glass/utils/NativeLibLoader.java

```diff
@@ -150,6 +173,60 @@ private static void loadLibraryInternal(String libraryName) {
         }
     }
 
+   /**
+    * If there is a library with the platform-correct name at the
+    * root of the resources in this jar, use that.
+    */
+    private static boolean loadLibraryFromResource(String libraryName, List<String> dependencies, Class caller) {
+        try {
+            // first preload dependencies
+            if (dependencies != null) {
+                for (String dep: dependencies) {
+                    loadLibraryFromResource(dep, null, caller);
```

Comment:
The mechanism generally looks like it works. However, I think should just unpack the dependencies and not load them. The media code loads only one of the N libavplugin-xxxx.so libraries, and does so from native code.

Reply:
If media code relies on the search path, not preloading them might not work on Windows because it doesn't support rpath.